Model/Details.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Aligent Consulting
+ * Copyright (c) Aligent Consulting (https://www.aligent.com.au)
+ */
+
+declare(strict_types=1);
+
+namespace Aligent\AsyncEvents\Model;
+
+use Aligent\AsyncEvents\Api\AsyncEventRepositoryInterface;
+use Aligent\AsyncEvents\Model\ResourceModel\AsyncEventLog\Collection;
+use Aligent\AsyncEvents\Model\ResourceModel\AsyncEventLog\CollectionFactory as AsyncEventLogCollectionFactory;
+
+class Details
+{
+    /**
+     * @var array
+     */
+    private $traceCache = [];
+
+    /**
+     * @var Collection
+     */
+    private $collection;
+
+    /**
+     * @var AsyncEventRepositoryInterface
+     */
+    private $asyncEventRepository;
+
+    public function __construct(
+        AsyncEventLogCollectionFactory $collectionFactory,
+        AsyncEventRepositoryInterface  $asyncEventRepository
+    ) {
+        $this->collection = $collectionFactory->create();
+        $this->asyncEventRepository = $asyncEventRepository;
+    }
+
+    public function getDetails($uuid)
+    {
+        if (array_key_exists($uuid, $this->traceCache)) {
+            return $this->traceCache[$uuid];
+        }
+
+        $this->collection->addFilter('uuid', $uuid);
+
+        $traces = $this->collection->toArray();
+
+        if ($traces['totalRecords'] === 0) {
+            return [];
```

Comment:
To avoid the undefined index issues I've mentioned elsewhere, you could return an array like:
```
['traces' => []]
```
(You'd also need to set to `$this->traceCache[$uuid]`)